test_plans/runtime_classes.asciidoc

```diff
@@ -0,0 +1,52 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for context class and device class
+
+This is a test plan for the context class as described in Section 4.6.3. of the SYCL 2020 specification and for the device class as described in Section 4.6.4. of the SYCL 2020 specification. This test plan adds coverage for the missing functionality that is part of SYCL 2020.
+
+Estimated development time is one day.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are not backend-specific and are performed for any SYCL backend.
+
+=== Device coverage
+
+All tests construct a test device for which conformance is assessed. All the tests described below are performed once for that test device.
+
+== Tests
+No negative test are included.
+
+=== Context
+
+==== Existing test
+The existing test in `context/context_info.cpp` checks the return value of `sycl::context.get_info` using `check_get_info_param` for various context information descriptors.
+
+==== Modification
+The test is modified to include the following context information descriptors that were added in SYCL 2020:
+
+- `info::context::atomic_memory_order_capabilities` Additionally, check that the returned list contains `sycl::memory_order::relaxed`.
+- `info::context::atomic_fence_order_capabilities` Additionally, check that the returned list contains `sycl::memory_order::relaxed`, `sycl::memory_order::acquire`, `sycl::memory_order::release`, and `sycl::memory_order::acq_rel`.
+- `info::context::atomic_memory_scope_capabilities` Additionally, check that the returned list contains `sycl::memory_scope::work_group`.
+- `info::context::atomic_fence_scope_capabilities` Additionally, check that the returned list contains `sycl::memory_scope::work_group`.
+
+=== Device
```

Comment:
Did you intend for this test plan to cover all of the new 2020 functionality added to the `device` class?  Note that the member function `device::has` is also new.  Should that be covered in this test plan, or is it covered elsewhere?

Reply:
> Did you intend for this test plan to cover all of the new 2020 functionality added to the device class?

Yes, all new 2020 functionality not specific to backends. 

The member function `device::has` will be called for all defined aspects in the test described by the [device selector test plan](https://github.com/KhronosGroup/SYCL-CTS/pull/422).